extras/module-node/README.rst

```diff
@@ -0,0 +1,62 @@
+=============================
+Node.js-like module framework
+=============================
+
+This directory contains an example module resolution and loading framework and
+``require`` implementation based on Node.js.  The application need only
+provide the module resolution and loading logic:
+
+* Add ``duk_module_node.c`` to list of C sources to compile.
+
+* Ensure ``duk_module_node.h`` is in the include path.
+
+* Include the extra header in calling code and initialize the bindings::
+
+      #include "duktape.h"
+      #include "duk_module_node.h"
+
+      /* After initializing the Duktape heap or when creating a new
+       * thread with a new global environment:
+       */
+      duk_module_node_init(ctx, resolve_cb, load_cb);
```

Comment:
If this convention was changed so that `duk_module_node_init()` would expect that callbacks in the value stack, it would then support Ecmascript resolve/load functions. If you want to make versioning easier, the value stack argument could also be an initialization parameter object providing the callbacks which would then be easier to extend:

    {
        "resolve": ...,
        "load": ...
    }